core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/template/TemplateValidator.java

```diff
@@ -157,12 +159,22 @@ private void validateVolumeType(VolumeTemplate value, Platform platform, Validat
         }
     }
 
+    private boolean isIDBrokerInstanceGroup(InstanceGroup instanceGroup) {
+        return GROUP_NAME_ID_BROKER.equalsIgnoreCase(instanceGroup.getGroupName());
+    }
+
     private void validateVolumeCount(VolumeTemplate value, VmType vmType, VolumeParameterType volumeParameterType,
-        ValidationResult.ValidationResultBuilder validationBuilder) {
+        ValidationResult.ValidationResultBuilder validationBuilder, InstanceGroup instanceGroup) {
         if (vmType != null && needToCheckVolume(volumeParameterType, value.getVolumeCount())) {
             VolumeParameterConfig config = vmType.getVolumeParameterbyVolumeParameterType(volumeParameterType);
             if (config != null) {
-                if (value.getVolumeCount() > config.maximumNumber()) {
+                if (isIDBrokerInstanceGroup(instanceGroup)) {
```

Comment:
Please do not duplicate code. There is no need for this `isIDBrokerInstanceGroup(instanceGroup)` branch. Instead, the original validation for the max limit can be reused without changes, and the extra check for the IDBroker instance group name can be added to the min limit validation below. The comment can be also moved in front of the whole `if` block, or possibly inside the min limit validation branch (whichever you prefer).